### ISSUE TITLE
Unable to connect to WiFi AP (203 - ASSOC_FAIL)

### ISSUE THREAD

**Issue by TrainzLuvr** [atanisoft/ESP32CommandStation#14]
I keep getting:
    `Reason: 203 - ASSOC_FAIL`
...and on and on retrying to connect with that error.

My AP is MAC locked, and I did add ESP32's MAC address into it. Does this actually pose a problem or the issue is somewhere else?

It would also be nice if the serial log displayed the MAC address for such occasions. I had to hack it in myself in the DCCppESP32.cpp :)

**Comment by atanisoft** [comment]
I'm not certain what might cause this specific error. I'll dig into the esp32 code and get you some details on how to turn on more logging, but in PlatformIO.ini make sure the log level is set to verbose. That may give a bit more details but may not, I suspect a call to esp_set_log_level may be needed for the IDF code

**Comment by atanisoft** [comment]
@TrainzLuvr add these in WiFiInterface.cpp just above WiFi.mode(WIFI_STA):
```
  esp_log_level_set("wifi", ESP_LOG_VERBOSE);
  esp_wifi_internal_set_log_level(WIFI_LOG_VERBOSE);
  esp_wifi_internal_set_log_mod(WIFI_LOG_MODULE_ALL, WIFI_LOG_SUBMODULE_ALL, true);
```
This will turn on a lot of extra details from the wifi module.

**Comment by TrainzLuvr** [comment]
You know, after all this it turned out it was my wireless router. #@)($%*!&

Despite it working fine for many years, it seems that today's devices aren't so kind to older hardware, and thus ESP32 just didn't like talking to it.

I spent the whole day Sunday on testing router firmwares, starting with the vendor latest firmware (from 2017, LOL) and then going through various forks, OpenWRT, Merlin landing finally on DD-WRT.

From little testing I did last night, it appears everything is working and ESP32 can connect to the router, and I can connect to the DCC++ESP32 web interface (nice work there :+1: ).

I'm going to hook up a motor shield to ESP32 now and try all this together. Perhaps I will be able to retire my original DCC++.
